File: MantenimientoEntidades/Equipo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Mantenimiento.Entidades
{
    public class Equipo
    {
        //EQUI_Interno int,
        //EQUI_Nombre varchar(80),
        //EQUI_Marca varchar(30),
        //EQUI_Modelo varchar(30),
        //EQUI_Serie varchar(20),
        //EQUI_Codigo varchar(10),
        //EQUI_AnioFabricacion int,
        //EQUI_AnioServicio int,
        //EQUI_Estado char(1),
        //EQUI_Descripcion varchar(200),
        //TIPO_Interno int

        #region Propiedades
        private int? _EQUI_Interno;
        public int? EQUI_Interno
        {
            get { return _EQUI_Interno; }
            set { _EQUI_Interno = value; }
        }
        
        private string _EQUI_Nombre;
        public string EQUI_Nombre
        {
            get { return _EQUI_Nombre; }
            set { _EQUI_Nombre = value; }
        }

        private string _EQUI_Marca;
        public string EQUI_Marca
        {
            get { return _EQUI_Marca; }
            set { _EQUI_Marca = value; }
        }

        private string _EQUI_Modelo;
        public string EQUI_Modelo
        {
            get { return _EQUI_Modelo; }
            set { _EQUI_Modelo = value; }
        }

        private string _EQUI_Serie;
        public string EQUI_Serie
        {
            get { return _EQUI_Serie; }
            set { _EQUI_Serie = value; }
        }

        private string _EQUI_Codigo;
        public string EQUI_Codigo
        {
            get { return _EQUI_Codigo; }
            set { _EQUI_Codigo = value; }
        }

        private int? _EQUI_AnioFabricacion;
        public int? EQUI_AnioFabricacion
        {
            get { return _EQUI_AnioFabricacion; }
            set { _EQUI_AnioFabricacion = value; }
        }

        private int? _EQUI_AnioServicio;
        public int? EQUI_AnioServicio
        {
            get { return _EQUI_AnioServicio; }
            set { _EQUI_AnioServicio = value; }
        }

        private string _EQUI_Estado;
        public string EQUI_Estado
        {
            get { return _EQUI_Estado; }
            set { _EQUI_Estado = value; }
        }

        private string _EQUI_Descripcion;
        public string EQUI_Descripcion
        {
            get { return _EQUI_Descripcion; }
            set { _EQUI_Descripcion = value; }
        }

        private int _TIPO_Interno;
        public int TIPO_Interno
        {
            get { return _TIPO_Interno; }
            set { _TIPO_Interno = value; }
        }

        private int? _LOCA_Interno;
        public int? LOCA_Interno
        {
            get { return _LOCA_Interno; }
            set { _LOCA_Interno = value; }
        }

        private string _LOCA_Nombre;
        public string LOCA_Nombre
        {
            get { return _LOCA_Nombre; }
            set { _LOCA_Nombre = value; }
        }

        private string _LOCA_NombreExtendido;
        public string LOCA_NombreExtendido
        {
            get { return _LOCA_NombreExtendido; }
            set { _LOCA_NombreExtendido = value; }
        }

        private DateTime? _HILO_Fecha;
        public DateTime? HILO_Fecha
        {
            get { return _HILO_Fecha; }
            set { _HILO_Fecha = value; }
        }

        private string _LOCA_ID;
        public string LOCA_ID
        {
            get { return _LOCA_ID; }
            set { _LOCA_ID = value; }
        }

        #endregion

        #region CONSTRUCTORES
        public Equipo() { }
        public Equipo(int EQUI_Interno)
        {
            this.EQUI_Interno = EQUI_Interno;
        }
        public Equipo(int? EQUI_Interno,string EQUI_Nombre,string EQUI_Marca,string EQUI_Modelo,string EQUI_Serie,
            string EQUI_Codigo, int? EQUI_AnioFabricacion,int? EQUI_AnioServicio,string EQUI_Estado,
            string EQUI_Descripcion, int TIPO_Interno)
        {
            this.EQUI_Interno = EQUI_Interno;
            this.EQUI_Nombre = EQUI_Nombre;
            this.EQUI_Marca = EQUI_Marca;
            this.EQUI_Modelo = EQUI_Modelo;
            this.EQUI_Serie = EQUI_Serie;
            this.EQUI_Codigo = EQUI_Codigo;
            this.EQUI_AnioFabricacion = EQUI_AnioFabricacion;
            this.EQUI_AnioServicio = EQUI_AnioServicio;
            this.EQUI_Estado = EQUI_Estado;
            this.EQUI_Descripcion = EQUI_Descripcion;
            this.TIPO_Interno = TIPO_Interno;
        }
        public Equipo(int? EQUI_Interno, string EQUI_Nombre, string EQUI_Marca, string EQUI_Modelo, string EQUI_Serie,
            string EQUI_Codigo, int? EQUI_AnioFabricacion, int? EQUI_AnioServicio, string EQUI_Estado,
            string EQUI_Descripcion, int TIPO_Interno, int? LOCA_Interno)
        {
            this.EQUI_Interno = EQUI_Interno;
            this.EQUI_Nombre = EQUI_Nombre;
            this.EQUI_Marca = EQUI_Marca;
            this.EQUI_Modelo = EQUI_Modelo;
            this.EQUI_Serie = EQUI_Serie;
            this.EQUI_Codigo = EQUI_Codigo;
            this.EQUI_AnioFabricacion = EQUI_AnioFabricacion;
            this.EQUI_AnioServicio = EQUI_AnioServicio;
            this.EQUI_Estado = EQUI_Estado;
            this.EQUI_Descripcion = EQUI_Descripcion;
            this.TIPO_Interno = TIPO_Interno;
            this.LOCA_Interno = LOCA_Interno;
        }
        public Equipo(int? EQUI_Interno, string EQUI_Nombre, string EQUI_Marca, string EQUI_Modelo, string EQUI_Serie,
            string EQUI_Codigo, int? EQUI_AnioFabricacion, int? EQUI_AnioServicio, string EQUI_Estado,
            string EQUI_Descripcion, int TIPO_Interno, int? LOCA_Interno, string LOCA_Nombre, string LOCA_NombreExtendido)
        {
            this.EQUI_Interno = EQUI_Interno;
            this.EQUI_Nombre = EQUI_Nombre;
            this.EQUI_Marca = EQUI_Marca;
            this.EQUI_Modelo = EQUI_Modelo;
            this.EQUI_Serie = EQUI_Serie;
            this.EQUI_Codigo = EQUI_Codigo;
            this.EQUI_AnioFabricacion = EQUI_AnioFabricacion;
            this.EQUI_AnioServicio = EQUI_AnioServicio;
            this.EQUI_Estado = EQUI_Estado;
            this.EQUI_Descripcion = EQUI_Descripcion;
            this.TIPO_Interno = TIPO_Interno;
            this.LOCA_Interno = LOCA_Interno;
            this.LOCA_Nombre = LOCA_Nombre;
            this.LOCA_NombreExtendido = LOCA_NombreExtendido;
        }
        #endregion
    }
}

File: AplicacionWeb/Scripts/Equipo.js
/// <reference path="jquery-2.0.0-vsdoc.js" />
//-------------------para el select de localizaciones
var localizaciones = null;
var charTab = "-";
var numCharTab = 1;
$(document).ready(function (e) {
    ObtenerLocalizaciones();
    if ($("#IDEquipo").val() != "" && $("#opcURL").val() == "editar") {
        $(".pagetitle h2").html("Datos de Equipo");
        $("#nuevo").parent("li").hide();
        $("#guardar_nuevo").parent("li").hide();
        ObtenerDatosDeEquipo();
    }
    $("#Nombre,#Marca,#Modelo,#Serie,#Codigo").change(function () {
        var descripcion = $("#Nombre").val() + " " +
                            $("#Marca").val() + " " +
                            $("#Modelo").val() + " " +
                            $("#Serie").val() + " " +
                            $("#Codigo").val();
        $("#Descripcion").val(descripcion);
    });
    DesmarcarCamposVaciosLLenados(); //desmarcamos los campos marcados que se van llenando
    CamposNumericos("#AnioFabricacion,#AnioServicio"); //mandamos los camppos que seran numericos


    $("#nuevo").click(function (e) {
        e.preventDefault();
        //alert("kk");
        $("#IDEquipo").val("");
        $("#HILO_Fecha").val("");
        limpiarMsjBox();
        limpiarTodosLosCampos();
    });
    $("#guardar").click(function (e) {
        e.preventDefault();
        if (CamposRequeridosVacios("#Nombre,#Estado,#Tipo,#Localizacion")) {//si estan vacios
            showError("Los campos marcados son obligatorios");
            return false;
        } else {
            limpiarMsjBox();
            quitarBgerror();
            if (GuardarEquipo() > 0) {
                ObtenerDatosDeEquipo();
            }

        }
    });
    $("#guardar_cerrar").click(function (e) {
        e.preventDefault();
        if (CamposRequeridosVacios("#Nombre,#Estado,#Tipo,#Localizacion")) {//si estan vacios
            showError("Los campos marcados son obligatorios");
            return false;
        } else {
            GuardarEquipo();
            //window.history.back();
            location.href = "GestionEquipos.aspx";
        }
    });
    $("#guardar_nuevo").click(function (e) {
        e.preventDefault();
        if (CamposRequeridosVacios("#Nombre,#Estado,#Tipo,#Localizacion")) {//si estan vacios
            showError("Los campos marcados son obligatorios");
            return false;
        } else {
            GuardarEquipo();
            //limpiarMsjBox();
            limpiarTodosLosCampos();
        }
    });

});  //fin document


//----------funciones de equipo
function GuardarEquipo() {
    var ID = 0;
    var equipo = new Object();
    equipo.EQUI_Interno = $("#IDEquipo").val();
    equipo.EQUI_Nombre = $("#Nombre").val();
    equipo.EQUI_Marca = $("#Marca").val();
    equipo.EQUI_Modelo = $("#Modelo").val();
    equipo.EQUI_Serie = $("#Serie").val();
    equipo.EQUI_Codigo = $("#Codigo").val();
    equipo.EQUI_AnioFabricacion = $("#AnioFabricacion").val();
    equipo.EQUI_AnioServicio = $("#AnioServicio").val();
    equipo.EQUI_Estado = $("#Estado").val();
    equipo.EQUI_Descripcion = $("#Descripcion").val();
    equipo.TIPO_Interno = $("#Tipo").val();
    equipo.LOCA_Interno = $("#Localizacion").val();
    equipo.HILO_Fecha = $("#HILO_Fecha").val();
    $.ajax({
        data: JSON.stringify(equipo),
        type: "POST",
        dataType: "json",
        contentType: 'application/json; charset=utf-8',
        url: "Equipo.aspx/GuardarEquipo",
        async: false,
        cache: false,
        success: function (data) {
            ID = parseInt(data.d);
            
            if (ID > 0) {
                showMensaje("Datos guardados correctamente");
                $("#IDEquipo").val(ID);
            } else {
                showError("No se insertó el usuario");
                //$("#IDEquipo").val("");
            }
        }//fin success
    }); //fin de ajax
    return ID;
} //fin guardar equipo
function ObtenerDatosDeEquipo() {
    var equipo = new Object();
    equipo.EQUI_Interno = $("#IDEquipo").val();
    $.ajax({
        data: JSON.stringify(equipo),
        type: "POST",
        dataType: "json",
        contentType: 'application/json; charset=utf-8',
        url: "Equipo.aspx/ObtenerEquipo",
        async: false,
        cache: false,
        success: function (data) {
            var item = data.d;

            $("#IDEquipo").val(item.EQUI_Interno);
            $("#Nombre").val(item.EQUI_Nombre);
            $("#Marca").val(item.EQUI_Marca);
            $("#Modelo").val(item.EQUI_Modelo);
            $("#Serie").val(item.EQUI_Serie);
            $("#Codigo").val(item.EQUI_Codigo);
            $("#AnioFabricacion").val(item.EQUI_AnioFabricacion);
            $("#AnioServicio").val(item.EQUI_AnioServicio);
            $("#Tipo").val(item.TIPO_Interno);
            $("#Estado").val(item.EQUI_Estado);
            $("#Descripcion").val(item.EQUI_Descripcion);
            $("#Localizacion").val(item.LOCA_Interno);
            if (item.HILO_Fecha != null)
                $("#HILO_Fecha").val(ConvertirDateTimeAString(item.HILO_Fecha, "dd/mm/yyyy HH:MM:ss.l"));
            else
                $("#HILO_Fecha").val("");
            //alert(ConvertirDateTimeAString(item.HILO_Fecha, "dd/mm/yyyy HH:MM:ss.l"));
            //alert(dateFormat(Date(), "dd/mm/yyyy HH:MM:ss"));
        }
    });  //fin de ajax
   
}

function ObtenerLocalizaciones() {

    $.ajax({
        type: "POST",
        dataType: "json",
        contentType: 'application/json; charset=utf-8',
        url: "Equipo.aspx/Localizaciones",
        async: false,
        cache: false,
        success: function (data) {
            //variable global localizaciones
            localizaciones = data.d;
        }
    });   //fin ajax
    //obtenemos el select arbol
    if (localizaciones != null) {

        ArbolSelectLocalizaciones(1);
    }
} //fin ObtenerLocalizaciones
function ArbolSelectLocalizaciones(padre) {
    var IDLoc = null;
    var NomLoc = null;
    $.each(localizaciones, function (i, item) {
        if (item.LOCA_Origen == padre) {
            var tab = "";
            for (i = 1; i <= numCharTab; i++)
                tab = tab + " " + charTab;
            $("#Localizacion").append("<option value='" + item.LOCA_Interno + "'>" + tab + item.LOCA_Nombre + "</option>");
            numCharTab++;
            ArbolSelectLocalizaciones(item.LOCA_Interno);
            numCharTab--;
        }
    });

}
